### DISCUSSION TITLE
Share payload between Gateway and Api Builder

### DISCUSSION THREAD

**Discussion by Rawane** [Axway-API-Management-Plus/apigateway-openlogging-elk#216] [Q&A]
Hi,
Our infrastructure does not allow us to share volumes between the gateway and the API builder?
Do you have a solution?
The gateways are on standard environment and the API builder on K8s

**Comment by cwiechmann** [comment]
Right now, the API-Builder application expects the payload files to be locally available and reads them directly from file system:
![image](https://user-images.githubusercontent.com/16292002/173318872-d402a03a-ed23-4583-9a64-7b795a3b3019.png)

An option would be to call another REST-API HTTP-Endpoint which returns the payload content. 
Such an API could be exposed by the API-Gateways itself using Policies, as they have the payload files anyway. Possible parameters for such an API would be:

- **payloadFile**: The folder and name of the payload file (e.g. 2022-06-13/09.27/f402a7626e0279075a1b987a-0-sent)
- **serviceID**: Which is the API-Gateway instance name (e.g. traffic-7cbb77bc9d-hd6wx or instance-1)
- **correlationID**: For reference the transaction ID

As the API-Builder already knows the topology, it knows which ServiceID (Gateway-Instance) is running on which machine. With that, it would send such an API-Request to the API-Gateway that should have the payload file.

Would that be an option? 

However, I cannot guarantee this can be implemented, as it's the solution becomes fully officially supported by Axway PM and Axway R&D and with that, they decide what to implement and what not.

**Comment by Rawane** [comment]
Yes we will study this option.
However our manager had mentioned seeing the S3 solution from Amazon if it is possible to send the logs through the S3 protocol.
Or another solution take out the builder api from kubernete and put it in the same platform as the gatway.
In summary :
1) the solution you propose in your last message
2) Using amazon S3 protocol.
3) Put the API builder in the same platform as the gateway.
what is your opinion on the 3 solutions?
